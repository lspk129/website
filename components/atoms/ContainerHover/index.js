import styled from 'styled-components';

const ContainerHover = styled.div`
  transition: all 0.5s ease-in-out;
  ${({ scale }) => scale && `
    img {
      width: 200px;
      height: 200px;
    }
    &:hover {
      transform: scale(${scale});
    }
  `}
  .icons {
    transition: all 0.5s ease-in-out;
    color: ${({ theme }) => theme.background};
  }
  ${({ backgroundColor, theme }) => backgroundColor && `
    &:hover {
      background: ${theme.background};
      color: ${theme.text};
      .icons {
        color: ${theme.text};
      }
    };
  `}
`;

export default ContainerHover;
