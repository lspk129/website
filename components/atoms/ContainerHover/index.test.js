import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ContainerHover from './';

describe('ContainerHover', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<ContainerHover />);
    expect(wrapper).toBePresent();
    expect(wrapper).toHaveTagName('div');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders scale boolean props', () => {
    const wrapper = shallow(<ContainerHover scale={2} />);
    expect(wrapper).toHaveProp('scale', 2);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders background color boolean props with theme props', () => {
    const wrapper = shallow(<ContainerHover backgroundColor />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
