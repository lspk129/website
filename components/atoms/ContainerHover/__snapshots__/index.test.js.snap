// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContainerHover renders background color boolean props with theme props 1`] = `
<div
  className="sc-bdVaJa fvVKxK"
/>
`;

exports[`ContainerHover renders correctly 1`] = `
<div
  className="sc-bdVaJa klJskJ"
/>
`;

exports[`ContainerHover renders scale boolean props 1`] = `
<div
  className="sc-bdVaJa esKmAd"
  scale={2}
/>
`;
